### ISSUE TITLE
Publisher confirm - ConfirmListener not removed

### ISSUE THREAD

**Issue by pmackowski** [reactor/reactor-rabbitmq#98]
[ConfirmListener](https://github.com/reactor/reactor-rabbitmq/blob/master/src/main/java/reactor/rabbitmq/Sender.java#L572) added in **Sender.PublishConfirmSubscriber** is not removed. 
It is not a problem when channel is internally managed by Sender because it is closed in [doFinally](https://github.com/reactor/reactor-rabbitmq/blob/master/src/main/java/reactor/rabbitmq/Sender.java#L169) and all listeners are gone.

@acogoluegnes In case of pooled channels or managed by client, wouldn't it be a problem?
If yes, I can provide a fix.

**Comment by acogoluegnes** [comment]
This is indeed a problem, the channel should go back to the pool clean. I remember the channel pooling benchmark you wrote showed bad performance for the publisher confirm case, I don't remember the exact lifecycle of the benchmark, but the accumulation of confirm listener instances in pooled channels could explain the degradation of performance.

Sure, a fix would be welcome.

**Comment by pmackowski** [comment]
I have prepared a fix [PR#99](https://github.com/reactor/reactor-rabbitmq/pull/99). Indeed, not releasing ConfirmListener was the cause of bad channel pooling performance. Here are some benchamrks:

**Without channel pool**
```
Benchmark                                                   (channelPoolSize)  (nbMessages)   Mode  Cnt     Score   Error  Units
SenderBenchmark.sendWithPublishConfirms                                   N/A             1  thrpt       1871.371          ops/s
SenderBenchmark.sendWithPublishConfirms                                   N/A            10  thrpt       1055.587          ops/s
SenderBenchmark.sendWithPublishConfirms                                   N/A           100  thrpt        229.392          ops/s
SenderBenchmark.sendWithPublishConfirms                                   N/A          1000  thrpt         23.858          ops/s
```

**Channel pool before fix**
```
Benchmark                                                   (channelPoolSize)  (nbMessages)   Mode  Cnt     Score   Error  Units
SenderWithLazyChannelPoolBenchmark.sendWithPublishConfirms                  5             1  thrpt       1206.364          ops/s
SenderWithLazyChannelPoolBenchmark.sendWithPublishConfirms                  5            10  thrpt        764.831          ops/s
SenderWithLazyChannelPoolBenchmark.sendWithPublishConfirms                  5           100  thrpt        209.655          ops/s
SenderWithLazyChannelPoolBenchmark.sendWithPublishConfirms                  5          1000  thrpt         23.963          ops/s
```

**Channel pool after fix**
```
Benchmark                                                   (channelPoolSize)  (nbMessages)   Mode  Cnt     Score   Error  Units
SenderWithLazyChannelPoolBenchmark.sendWithPublishConfirms                  5             1  thrpt       5411.111          ops/s
SenderWithLazyChannelPoolBenchmark.sendWithPublishConfirms                  5            10  thrpt       1460.724          ops/s
SenderWithLazyChannelPoolBenchmark.sendWithPublishConfirms                  5           100  thrpt        240.917          ops/s
SenderWithLazyChannelPoolBenchmark.sendWithPublishConfirms                  5          1000  thrpt         23.784          ops/s
```

**Comment by acogoluegnes** [comment]
Fixed by #99.

**Comment by acogoluegnes** [comment]
@pmackowski Thanks for confirming that not removing the confirm listener was the cause of poor performance.
